lib/pliny/middleware/canonical_log_line.rb

```diff
@@ -0,0 +1,121 @@
+module Pliny::Middleware
+  # Emits a "canonical log line", i.e. a single log line that contains as much
+  # relevant information about a request as possible and which makes for a
+  # single convenient reference point to understand all the vitals of any
+  # single request.
+  #
+  # This default implementation contains some useful data to get a project
+  # started, but it's usually recommended to vendor this middleware into your
+  # project and start adding some custom fields. Some examples of those might
+  # be:
```

Comment:
@gudmundur Cool, yeah I would be too opposed to this.

What would you think about the `LogLine` class living in user-land but including a default module from Pliny, like:

``` ruby
class LogLine
  include Pliny::CanonicalLogLineHelpers
  include Pliny::CanonicalLogLineDefaultFields

  ...
```

The vendored middleware would then contain something like this:

``` ruby
def call(env)
  begin
    start = Time.now
    status, headers, response = @app.call(env)
  ensure
    begin
      line = LogLine.new
      Pliny::CanonicalLogLineDefaultFields.extract(line, status, headers, response,
        start: start)
      ...
```

That way we could still tweak the default fields while allowing some extensibility right away.

The other plausible option is just to vendor it completely. I think you could make the argument for that because it just makes things so much simpler and easier to look at, but at the cost of possibly missing out on bug fixes and updates.